kex-runner/src/main/kotlin/org/vorpal/research/kex/jacoco/minimization/GreedyAlgorithm.kt

```diff
@@ -0,0 +1,64 @@
+package org.vorpal.research.kex.jacoco.minimization
+
+import java.nio.file.Path
+
+interface TestReduction
+
+class Test(
+    val reqs: List<Int>
+) {
+    var power: Int = reqs.size
+}
+
+class Requirement {
+    private val satisfyTests: MutableList<Path> =mutableListOf<Path>()
+    private var visited = false
+
+    fun addTest(testName: Path) {
+        satisfyTests.add(testName)
+    }
+
+    fun visit(tests: MutableMap<Path, Test>) {
+        if (visited) return
+        satisfyTests.forEach{ tests[it]!!.power-- }
+        visited = true
+    }
+}
+
+class GreedyAlgorithm(
```

Comment:
I would also rename this to `GreedyTestReductor` or `GreedyTestReductionImpl` so that it is more descriptive